demos/common/samples/ocv_common.hpp

```diff
@@ -27,7 +27,7 @@ void matU8ToBlob(const cv::Mat& orig_image, InferenceEngine::Blob::Ptr& blob, in
     if (static_cast<size_t>(orig_image.channels()) != channels) {
         THROW_IE_EXCEPTION << "The number of channels for net input and image must match";
     }
-    T* blob_data = blob->buffer().as<T*>();
+    T* blob_data = InferenceEngine::as<InferenceEngine::MemoryBlob>(blob)->rwmap().as<T*>();
```

Comment:
Pretty sure you can't do that. The pointer returned from `as` will become invalid as soon as the object returned by `rwmap` is destroyed.

Reply:
@IRDonch,

Can I fix it like this? Would you approve?
Ofc I'll fix other demos too.
```
InferenceEngine::LockedMemory<void> mappedBlob = InferenceEngine::as<InferenceEngine::MemoryBlob>(blob)->rwmap();
T* blob_data = mappedBlob.as<T*>();
```

Reply:
Sure, just watch the object lifetimes. You need to make sure that all blobs are unmapped before any inference happens.

I would also suggest using `auto` for the `LockedMemory` object, otherwise the code gets really cumbersome.

Reply:
I disagree about `auto`. `LockedMemory` makes it clearer what is returned by `rmap()`. I think `auto` should be used when it is obvious what the actual type is. In this case `auto` makes a reader find the declaration of `rmap()`. While demos don't focus on on IE API, I believe it still will be easier to read with `LockedMemory`.

Reply:
OK. Doesn't matter that much to me.